## Call Centers

### Introduction

In this project we use a dataset from [data.world](https://data.world/markbradbourne/rwfd-real-world-fake-data/workspace/file?filename=Call+Center.csv) website. The dataset we're going to use is from 'RWFD Real World Fake Data' by Mark Bradbourne, he have some datasets that we can use for free, and i picked one the call-center dataset.

The call-center dataset that we're going to use has 32,941 records of data that provide information about calls made to various call centers in North America. It includes the ID of the call , the customer name who called and also their reason, how long did the calls last in duration minutes, etc, that i'll use to explore the data.

### Importing data

After downloading the call center data, we need to import it to our relational database system, in this case i will use postgresql.

First we need to create a database inside the postgresql, i created mine as a project1, and in this database we are going to import our dataset. To import our dataset, we ned to create a table using a 'CREATE TABLE' syntax and next you write the table name 'calls', and then write the columns name with the data types and the size of the variable that will be in our 'calls' table. In the existing call center dataset we have 12 columns that include, the id, customer name, sentiment, city, reason, etc. So we are going to write the 12 columns in our database so it will be match with the existing call center dataset. Make sure to write the columns sequentially according to the call center datasets as the existing data. 

You can see or follow along the steps [here](https://github.com/tinashdj/Call-Center-Exploratory-Data-Analysis/blob/main/datasets/createdbs.sql). 

### Exploratory Data Analysis

Exploratory Data Analysis (EDA) is the process of visualizing and analyzing data to extract insights from it. In other words, EDA is the process of summarizing important characteristics of data in order to gain better understanding of the dataset.
[Source](https://medium.com/code-heroku/introduction-to-exploratory-data-analysis-eda-c0257f888676)

### 1. How many call centers?

Lets see how many call centers do we have in the call center dataset

In [7]:
%%sql 

SELECT DISTINCT call_center FROM calls

 * postgresql://postgres:***@localhost/Project1
4 rows affected.


[('Los Angeles/CA      ',),
 ('Chicago/IL          ',),
 ('Denver/CO           ',),
 ('Baltimore/MD        ',)]

The results we have only 4 call centers which is in Los Angeles/CA, Chicago/IL, Denver/CO and Baltimore/MD.

### 2. What is the most reason of the calls?

From the over 32.900 calls data, we want to know what is the reason of the calls and which one is be the most reason. 

In [8]:
%%sql

SELECT reason, COUNT(*),
ROUND(COUNT(*) * 100.0/ (SELECT COUNT(*) FROM calls) ,2) AS percentage
FROM calls GROUP BY reason ORDER BY reason ASC;

 * postgresql://postgres:***@localhost/Project1
3 rows affected.


[('Billing Question    ', 23462, Decimal('71.22')),
 ('Payments            ', 4749, Decimal('14.42')),
 ('Service Outage      ', 4730, Decimal('14.36'))]

We get 3 reasons over the calls, and the most reason of the calls is about Billing Question with 71.22% and 23.462 total calls.
Then we have still 2 reasons, Payments and Service Outage that really have a huge different total from the first one. Payments reason have 14.42% with 4.749 total of calls and it's almost have the same total with Service outage reason which have 14.36% with 4.730 total calls.

### 3. Where is the city has the most calls and what is the reason?

Let's see what city has the most calls and also the reason.

In [9]:
%%sql 

SELECT city, reason, COUNT(*) as total_calls
FROM calls 
GROUP BY city,reason ORDER BY total_calls DESC;

 * postgresql://postgres:***@localhost/Project1
1348 rows affected.


[('Washington                                        ', 'Billing Question    ', 806),
 ('Houston                                           ', 'Billing Question    ', 494),
 ('New York City                                     ', 'Billing Question    ', 407),
 ('El Paso                                           ', 'Billing Question    ', 374),
 ('Dallas                                            ', 'Billing Question    ', 312),
 ('Atlanta                                           ', 'Billing Question    ', 295),
 ('Miami                                             ', 'Billing Question    ', 259),
 ('Los Angeles                                       ', 'Billing Question    ', 243),
 ('Kansas City                                       ', 'Billing Question    ', 231),
 ('Sacramento                                        ', 'Billing Question    ', 226),
 ('Chicago                                           ', 'Billing Question    ', 224),
 ('Springfield                                       ', 'Billing Question    ', 219),
 ('San Antonio                                       ', 'Billing Question    ', 217),
 ('Oklahoma City                                     ', 'Billing Question    ', 205),
 ('Austin                                            ', 'Billing Question    ', 203),
 ('Phoenix                                           ', 'Billing Question    ', 201),
 ('Richmond                                          ', 'Billing Question    ', 197),
 ('Birmingham                                        ', 'Billing Question    ', 196),
 ('Minneapolis                                       ', 'Billing Question    ', 191),
 ('Cincinnati                                        ', 'Billing Question    ', 190),
 ('Las Vegas                                         ', 'Billing Question    ', 190),
 ('San Francisco                                     ', 'Billing Question    ', 188),
 ('Seattle                                           ', 'Billing Question    ', 186),
 ('Denver                                            ', 'Billing Question    ', 186),
 ('Pittsburgh                                        ', 'Billing Question    ', 184),
 ('Charlotte                                         ', 'Billing Question    ', 184),
 ('New Orleans                                       ', 'Billing Question    ', 183),
 ('Philadelphia                                      ', 'Billing Question    ', 182),
 ('San Diego                                         ', 'Billing Question    ', 180),
 ('Memphis                                           ', 'Billing Question    ', 178),
 ('Saint Louis                                       ', 'Billing Question    ', 166),
 ('Washington                                        ', 'Payments            ', 164),
 ('Tampa                                             ', 'Billing Question    ', 162),
 ('Tulsa                                             ', 'Billing Question    ', 160),
 ('Orlando                                           ', 'Billing Question    ', 159),
 ('Salt Lake City                                    ', 'Billing Question    ', 157),
 ('Portland                                          ', 'Billing Question    ', 156),
 ('Louisville                                        ', 'Billing Question    ', 155),
 ('Indianapolis                                      ', 'Billing Question    ', 155),
 ('Des Moines                                        ', 'Billing Question    ', 155),
 ('Columbus                                          ', 'Billing Question    ', 154),
 ('Charleston                                        ', 'Billing Question    ', 152),
 ('Boston                                            ', 'Billing Question    ', 144),
 ('Detroit                                           ', 'Billing Question    ', 143),
 ('Rochester                                         ', 'Billing Question    ', 140),
 ('Fresno                                            ', 'Billing Question    ', 140),
 ('Washington                               

So from the previous question we got that Billing Question reason is the most reason over the calls. From the result above, we can see Washington is the city that have the most total calls over the Billing Question reason with 806 total calls. And also it happened in the other some city that Billing Question be the most reasons of the calls, such as Houston with 494 total calls, and New York City with 407 total calls.

### 4. Which day has the most calls?

Next lets find out which day is the bussiest to receive the calls

In [10]:
%%sql

SELECT To_Char(call_timestamp, 'Day') as Day_of_call, COUNT(*) AS count_of_calls 
FROM calls GROUP BY Day_of_call ORDER BY count_of_calls DESC;

 * postgresql://postgres:***@localhost/Project1
7 rows affected.


[('Friday   ', 6546),
 ('Thursday ', 5536),
 ('Tuesday  ', 5432),
 ('Wednesday', 4556),
 ('Monday   ', 4307),
 ('Saturday ', 3314),
 ('Sunday   ', 3250)]

We get that Friday has the most calls with 6.548 total calls while sunday has the least with 3.250 total calls.

### 5. Maximal, Minimal and the Average of the calls duration

Now we want to know how long is the maximum call duration, minimum call duration and average call duration. We can find the answer by using aggregate function.

In [11]:
%%sql

SELECT MAX (call_duration_minutes) AS max_call_duration_, 
MIN (call_duration_minutes) AS min_call_duration_,
ROUND (AVG(call_duration_minutes),2) AS avg_call_duration_
FROM calls;

 * postgresql://postgres:***@localhost/Project1
1 rows affected.


[(45, 5, Decimal('25.02'))]

By using the MAX, MIN, and AVG aggregate function, we have that the maximum call duration lasts in 45 minutes, minimum call duration in 5 minutes, and the average is 25.02 minutes.

### 6. Response time by call centers

So the response time is the amount of time it takes from when the caller dials into the contact center, to the time they are connected with the appropriate agent.

In [12]:
%%sql

SELECT call_center, response_time, COUNT(*) AS count
FROM calls GROUP BY call_center, response_time 
ORDER BY call_center, count DESC;

 * postgresql://postgres:***@localhost/Project1
12 rows affected.


[('Baltimore/MD        ', 'Within SLA          ', 6855),
 ('Baltimore/MD        ', 'Below SLA           ', 2768),
 ('Baltimore/MD        ', 'Above SLA           ', 1389),
 ('Chicago/IL          ', 'Within SLA          ', 3361),
 ('Chicago/IL          ', 'Below SLA           ', 1361),
 ('Chicago/IL          ', 'Above SLA           ', 697),
 ('Denver/CO           ', 'Within SLA          ', 1741),
 ('Denver/CO           ', 'Below SLA           ', 692),
 ('Denver/CO           ', 'Above SLA           ', 343),
 ('Los Angeles/CA      ', 'Within SLA          ', 8668),
 ('Los Angeles/CA      ', 'Below SLA           ', 3327),
 ('Los Angeles/CA      ', 'Above SLA           ', 1739)]

A service level agreement (SLA) is a promise a contact center makes to provide a certain level of service to its clients. The most common SLA for a call center to make is committing to answering a percentage of inbound calls within a set time frame. From the result above, we can see how many calls are within, below and above the service level management time. For example Baltimore/MD call center has 6.855 calls within SLA, Chicago/IL call center has 1.361 calls below SLA, Denver/CO call center has 343 calls above SLA and Los Angeles/CA call center has 8.668 calls within SLA.

### 7. How many channels?

Channels refer to the method or communication channel a customer uses to interact with a business.

In [13]:
%%sql

SELECT DISTINCT channel
FROM calls;

 * postgresql://postgres:***@localhost/Project1
4 rows affected.


[('Chatbot             ',),
 ('Web                 ',),
 ('Email               ',),
 ('Call-Center         ',)]

So we have 4 channels that customers use to interact with. Such as chatbot, web, email and call center.

### 8. What is the most used channel by state?

So in our 'calls' database we have 'state' data field, we can use it to find the most used channel by state.

In [14]:
%%sql

SELECT state, channel, COUNT(*) AS count
FROM calls
GROUP BY state, channel ORDER BY count DESC
LIMIT 10;

 * postgresql://postgres:***@localhost/Project1
10 rows affected.


[('Texas               ', 'Call-Center         ', 1137),
 ('California          ', 'Call-Center         ', 1092),
 ('California          ', 'Chatbot             ', 962),
 ('Florida             ', 'Call-Center         ', 923),
 ('Texas               ', 'Chatbot             ', 899),
 ('California          ', 'Email               ', 831),
 ('Texas               ', 'Email               ', 805),
 ('California          ', 'Web                 ', 746),
 ('Texas               ', 'Web                 ', 731),
 ('Florida             ', 'Chatbot             ', 694)]

It looks like that call center channel is the most used channel. And the state that used it the most is in Texas with 1.137 total calls. And based on the result we also know that after Texas there's California with 1.092 total calls as the second state that used call center channel. In addition, California is also a state that uses the most channels other than call centers, such as chatbot channel with 962 total calls, email channel with 831 total calls and web channel with 746 total calls.

### 9. What is the most sentiment?

Let's see the total sentiment given by the customer based on the sentiment

In [15]:
%%sql

SELECT sentiment, COUNT(*),
ROUND(COUNT(*) * 100.0/ (SELECT COUNT(*) FROM calls) ,2) AS percentage
FROM calls GROUP BY sentiment ORDER BY sentiment ASC;

 * postgresql://postgres:***@localhost/Project1
5 rows affected.


[('Negative            ', 11063, Decimal('33.58')),
 ('Neutral             ', 8754, Decimal('26.57')),
 ('Positive            ', 3928, Decimal('11.92')),
 ('Very Negative       ', 6026, Decimal('18.29')),
 ('Very Positive       ', 3170, Decimal('9.62'))]

According to the result we get that Negative Sentiment is the sentiment that has been given the most by customers. The Negative sentiment has 11.063 total or 33.58%. Whereas Very Positive is the least sentiment with 3.170 total or 9.62%.

### 10. Who has given the 'Very Positive' sentiments?

After got the sentiments total, let's find out the customer name that gives 'Very Positive' sentiment.

In [16]:
%%sql

SELECT customer_name, sentiment
FROM calls
WHERE sentiment = 'Very Positive'
GROUP BY sentiment, customer_name
LIMIT 10;

 * postgresql://postgres:***@localhost/Project1
10 rows affected.


[('Cherice Coultas                                   ', 'Very Positive       '),
 ('Tome Grinnell                                     ', 'Very Positive       '),
 ('Billi Sulman                                      ', 'Very Positive       '),
 ('Alida Lovelock                                    ', 'Very Positive       '),
 ('Prudi Columbell                                   ', 'Very Positive       '),
 ('Norrie Eburne                                     ', 'Very Positive       '),
 ('Jose Cammis                                       ', 'Very Positive       '),
 ('Tansy Luggar                                      ', 'Very Positive       '),
 ('Roobbie Shingler                                  ', 'Very Positive       '),
 ('Celinda Fazackerley                               ', 'Very Positive       ')]

So we have 10 customers name that gave 'Very Positive' sentiment, but in this case we only take 10 customers.